src/rules/store/select-in-selector-names.ts

```diff
@@ -0,0 +1,76 @@
+import path from 'path'
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils'
+import { isIdentifier, isLiteral } from '../../utils/helper-functions/index'
+
+import { memorizedSelector, selectorsNames, docsUrl } from '../../utils'
+
+export const messageId = 'selectInSelectorNames'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: path.parse(__filename).name,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      category: 'Best Practices',
+      description: 'Name Selector Functions as selectThing',
+      recommended: 'warn',
+    },
+    schema: [],
+    messages: {
+      [messageId]: 'Name Selector Functions as selectThing',
+    },
+  },
+  defaultOptions: [],
+  create: (context) => {
+    const includedKeyword = 'select'
+
+    function hasKeyword(name: string): boolean {
+      // console.log(name.toLowerCase().startsWith(includedKeyword))
+      return name.toLowerCase().startsWith(includedKeyword)
+    }
+
+    return {
+      [`VariableDeclaration[declarations.length>0]`](
+        node: TSESTree.VariableDeclaration,
+      ) {
+        for (const dec of node.declarations) {
```

Comment:
I think we can query for `VariableDeclarator` (instead of `VariableDeclaration[declarations.length>0]`).
This way, we don't have to iterate over de declarators.

Reply:
To be honest, you can just write a selector to avoid both the iteration mentioned above and conditionals:

```ts
create(context) {
  return {
    'VariableDeclarator[id.name=/^(?!select).+/]:matches(:has(TSTypeAnnotation[typeAnnotation.typeName.name=/^MemoizedSelector(Props$|$)/]), :has(CallExpression[callee.name=/^(createSelector|createFeatureSelector)$/]))'({
      id,
    }: TSESTree.VariableDeclarator) {
      context.report({ node: id, messageId })
    },
  }
},
```

Reply:
And here is our query wizard 🪄

Reply:
![mind-blown](https://media4.giphy.com/media/wbcMnfHqOJX9K/200.gif)